pkg/bindinfo/binding.go

```diff
@@ -155,42 +131,32 @@ func prepareHints(sctx sessionctx.Context, binding *Binding) (rerr error) {
 	return nil
 }
 
-// `merge` merges two Bindings. It will replace old bindings with new bindings if there are new updates.
-func merge(lBindings, rBindings []*Binding) []*Binding {
-	if lBindings == nil {
-		return rBindings
-	}
-	if rBindings == nil {
-		return lBindings
-	}
-	result := lBindings
-	for i := range rBindings {
-		rbind := rBindings[i]
-		found := false
-		for j, lbind := range lBindings {
-			if lbind.isSame(rbind) {
-				found = true
-				if rbind.UpdateTime.Compare(lbind.UpdateTime) >= 0 {
-					result[j] = rbind
-				}
-				break
-			}
-		}
-		if !found {
-			result = append(result, rbind)
+func pickCachedBinding(bindings ...*Binding) *Binding {
+	// filter nil pointers in the binding slice
+	n := 0
+	for _, b := range bindings {
+		if b != nil {
+			bindings[n] = b
+			n++
 		}
 	}
-	return result
-}
+	bindings = bindings[:n]
 
-func removeDeletedBindings(br []*Binding) []*Binding {
-	result := make([]*Binding, 0, len(br))
-	for _, binding := range br {
-		if binding.Status != deleted {
-			result = append(result, binding)
+	if len(bindings) == 0 {
+		return nil
+	}
+	sort.Slice(bindings, func(i, j int) bool {
```

Comment:
use ```slices.SortFunc``` to avoid bound check

Reply:
I updated this implementation, now there is no Sort, PTAL